book/src/pong-tutorial/pong-tutorial-05.md

```diff
@@ -36,42 +34,57 @@ Then, we'll create `systems/winner.rs`:
 # }
 # 
 use amethyst::{
-    core::transform::Transform,
-    ecs::{System, World},
+    core::{
+        ecs::{ParallelRunnable, System},
+        transform::Transform,
+    },
+    ecs::{IntoQuery, SystemBuilder},
 };
 
 use crate::pong::{Ball, ARENA_HEIGHT, ARENA_WIDTH};
 
 pub struct WinnerSystem;
 
+// NEW
 impl System for WinnerSystem {
-    type SystemData = (.write_component::<Ball>() .write_component::<Transform>);
-
-    fn run(&mut self, (mut balls, mut locals): Self::SystemData) {
-        for (ball, transform) in (&mut balls, &mut locals).join() {
-            let ball_x = transform.translation().x;
-
-            let did_hit = if ball_x <= ball.radius {
-                // Right player scored on the left side.
-                println!("Player 2 Scores!");
-                true
-            } else if ball_x >= ARENA_WIDTH - ball.radius {
-                // Left player scored on the right side.
-                println!("Player 1 Scores!");
-                true
-            } else {
-                false
-            };
-
-            if did_hit {
-                ball.velocity[0] = -ball.velocity[0]; // Reverse Direction
-                transform.set_translation_x(ARENA_WIDTH / 2.0); // Reset Position
-                transform.set_translation_y(ARENA_HEIGHT / 2.0); // Reset Position
-            }
-        }
+    fn build(self) -> Box<dyn ParallelRunnable> {
+        Box::new(
+            SystemBuilder::new("WinnerSystem")
+                .with_query(<(&mut Ball, &mut Transform)>::query())
+                .write_component::<Ball>()
+                .write_component::<Transform>()
+                .build(
+                    move |_commands,
+                          world,
+                          (),
+                          query| {
+                        let (mut ball_world, mut _remaining) = world.split_for_query(query);
```

Comment:
I don't understand the purpose of `split_for_query` and I think it would be helpful to explain it here.